commands/command_track.go

```diff
@@ -221,8 +227,34 @@ ArgsLoop:
 	}
 }
 
+type PatternData struct {
+	Pattern  string `json:"pattern"`
+	Source   string `json:"source"`
+	Lockable bool   `json:"lockable"`
+	Tracked  bool   `json:"tracked"`
+}
+
 func listPatterns() {
 	knownPatterns := getAllKnownPatterns()
+	if trackJSONFlag {
+		patterns := struct {
+			Patterns []PatternData `json:"patterns"`
+		}{Patterns: make([]PatternData, 0, len(knownPatterns))}
+		for _, p := range knownPatterns {
+			patterns.Patterns = append(patterns.Patterns, PatternData{
+				Pattern:  p.Path,
+				Source:   p.Source.String(),
+				Tracked:  p.Tracked,
+				Lockable: p.Lockable,
+			})
+		}
+		err := json.NewEncoder(os.Stdout).Encode(patterns)
```

Comment:
Would it be possible to use the `SetIndent("", "  ")` [function](https://pkg.go.dev/encoding/json#Encoder.SetIndent) here to make the output format have a bit of indenting?

Reply:
Sure.  I doubt this is going to be very interesting for human-readable output, but it does at least make the tests nicer, so I figure that's an improvement.